### DISCUSSION TITLE
Fctx5 removes Russian layout after updating it's settings or on logout / reboot

### DISCUSSION THREAD

**Discussion by birdkiwi** [fcitx/fcitx5#1060] [Q&A] [answered]
Hi!

I have a problem with FCTX5 on Wayland session in Manjaro with Plasma 6.
I setup english and russian layouts: 
![image](https://github.com/fcitx/fcitx5/assets/3119744/f79f38bf-03cf-4201-b2fc-e0161530a076)

I have Pinyin input method as on the picture:
![image](https://github.com/fcitx/fcitx5/assets/3119744/d3c5999d-ad77-4a02-b3d6-b7023521f2c8)

Everything works fine, but after reboot / logout my russian layout disappears from keyboard layout settings, so I cannot switch between english and russian anymore.

I've found out that it happens also after fctx settings are changed.  

I thought changing XCB addon settings could help, but it doesn't change anything:
![image](https://github.com/fcitx/fcitx5/assets/3119744/df7e9475-b6ca-4be7-b877-f1f1b86663bc)

Could you guide me how to stop fctx override my keyboard layout settings after reboot / logout?

**Fctx5: 5.1.9**

Operating System: Manjaro Linux 
KDE Plasma Version: 6.0.4
KDE Frameworks Version: 6.1.0
Qt Version: 6.7.0
Kernel Version: 6.9.0-1-MANJARO (64-bit)
Graphics Platform: Wayland

**Comment by wengxt** [comment]
I’d suggest you add ru layout via fcitx, have you tried that?

**Reply by birdkiwi** [reply]
Thanks for the quick reply. Yes, I've tried that, I've created a separate group for Russian input method and set Win+Space shortcut to switch between en/pinyin and Russian groups. It works, but it has many problems with X applications, for example, switching group shortcut stops working after some time. 

I think the ideal solution here is to stop fctx5 from messing with global keyboard layout settings, but I don't know how. Maybe you have a better suggestion.

**Reply by wengxt** [reply, marked as answer]
Can you explain "Stops working after some time" part? Does it do nothing, or does it do something wrong? Fcitx 5 still uses xcb keygrab as the global hotkey detection, this doesn't work on wayland unless you change it in kcm_kwinxwayland to allow it listen to the keyboard event.

As for your original question, since you're using wayland you can change the corresponding option in wayland addon.
But it seems that I forget to add Configurable=True to the addon config so it doesn't show up in the configtool.

I just send a fix for that https://github.com/fcitx/fcitx5/commit/1ff448f3d98383ab7339756723cf52294e1e2f1a

You can put file ~/.config/fcitx5/conf/wayland.conf with content

```
Allow Overriding System XKB Settings=False
```

Or this command should also work

```
gdbus call    --session --dest org.fcitx.Fcitx5   \
   --object-path /controller \
  --method org.fcitx.Fcitx.Controller1.SetConfig \
  fcitx://config/addon/wayland "<{'Allow Overriding System XKB Settings': <'False'>}>"
```

**Comment by birdkiwi** [comment]
> Can you explain "Stops working after some time" part? Does it do nothing, or does it do something wrong?

Sorry, I haven't tested it properly. I just noticed strange behavior, it stopped working in Slack and Discord for some reason, after I press Win+Space (binded to change input groups) nothing happened in the input.  
 
> You can put file ~/.config/fcitx5/conf/wayland.conf with content
> 
> ```
> Allow Overriding System XKB Settings=False
> ```

This worked perfectly, that's what I need :) 谢谢你！
